test/integration/auth/rbac_test.go

```diff
@@ -680,3 +683,66 @@ func TestBootstrapping(t *testing.T) {
 	}
 	t.Errorf("error bootstrapping roles: %s", string(healthBytes))
 }
+
+func TestDiscoveryUpgradeBootstrapping(t *testing.T) {
+	var tearDownFn kubeapiservertesting.TearDownFunc
+	defer func() {
+		if tearDownFn != nil {
+			tearDownFn()
+		}
+	}()
+
+	// Pre-populate etcd with the old default RBAC discovery bindings
+	etcdConfig := framework.SharedEtcd()
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, nil, etcdConfig)
+	tearDownFn = server.TearDownFn
+
+	client, err := externalclientset.NewForConfig(server.ClientConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	t.Logf("Creating default `system:discovery` ClusterRole")
+	if _, err = client.RbacV1().ClusterRoles().Create(&rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{Name: "system:discovery"},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1helpers.NewRule("get").URLs(
+				"/healthz", "/version", "/version/",
+				"/openapi", "/openapi/*",
+				"/api", "/api/*",
+				"/apis", "/apis/*",
+			).RuleOrDie(),
+		},
+	}); err != nil {
+		t.Fatalf("Failed to create `system:discovery` ClusterRole: %v", err)
+	}
+
+	t.Logf("Creating default `system:discovery` ClusterRoleBinding")
+	binding := rbacv1helpers.NewClusterBinding("system:discovery").Groups(user.AllAuthenticated).BindingOrDie()
+	if _, err = client.RbacV1().ClusterRoleBindings().Create(&binding); err != nil {
+		t.Fatalf("Failed to create `system:discovery` ClusterRoleBinding: %v", err)
+	}
+
+	t.Logf("Creating default `system:basic-user` ClusterRole")
+	if _, err = client.RbacV1().ClusterRoles().Create(&rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{Name: "system:basic-user"},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1helpers.NewRule("create").Groups("authorization.k8s.io").Resources("selfsubjectaccessreviews", "selfsubjectrulesreviews").RuleOrDie(),
+		},
+	}); err != nil {
+		t.Fatalf("Failed to create `system:basic-user` ClusterRole: %v", err)
+	}
+
+	t.Logf("Creating default `system:basic-user` ClusterRoleBinding")
+	binding = rbacv1helpers.NewClusterBinding("system:basic-user").Groups(user.AllAuthenticated).BindingOrDie()
+	if _, err = client.RbacV1().ClusterRoleBindings().Create(&binding); err != nil {
+		t.Fatalf("Failed to create `system:basic-user` ClusterRoleBinding: %v", err)
+	}
+
+	// There is no guarantee on how long it takes the apiserver to honor the configuration and there is
```

Comment:
This is true for webhooks which is what I was testing, is this also true for the changes here?

Reply:
This is simply copy-pasting most of your code and running a sanity check to verify that I didn't break something (I can't get the test to run successfully locally). This test is definitely a work in progress :)

Reply:
I think it might run locally if you run it with `make test-integration`

Reply:
nope :/ I think that I'm going to have to spend more time debugging my etcd setup